qiskit/algorithms/optimizers/spsa.py

```diff
@@ -617,7 +617,7 @@ def minimize(
                 if self.termination_checker(
                     self._nfev, x_next, fx_check, np.linalg.norm(update), True
                 ):
-                    logger.info("terminated optimization at {k}/{self.maxiter} iterations")
+                    logger.info(f"terminated optimization at {k}/{self.maxiter} iterations")
```

Comment:
Thanks for noticing this. However this should be done similar to the logging statement just below that passes the substituted variables as parameters. This way when the call the logger.info is made, which will simply return if info level logging is not enabled, the formatted string does not have to be built each time just to be discarded if logging is not enabled. This is why the build failed, i.e. 
```
========================== Starting Command Output ===========================
/usr/bin/bash /home/vsts/work/_temp/a5311c74-6a01-4fae-b571-ed2c445bb1a2.sh
************* Module qiskit.algorithms.optimizers.spsa
qiskit/algorithms/optimizers/spsa.py:620:20: W1203: Use lazy % formatting in logging functions (logging-fstring-interpolation)
```